elasticdl/pkg/ps/server.go

```diff
@@ -2,52 +2,119 @@ package ps
 
 import (
 	"context"
+	"elasticdl.org/elasticdl/pkg/common"
 	pb "elasticdl.org/elasticdl/pkg/proto"
+	"fmt"
 	empty "github.com/golang/protobuf/ptypes/empty"
 	"google.golang.org/grpc"
 	"log"
 	"net"
+	"sync"
 )
 
-type psServer struct {
+// Server defines servicer of ps
+type Server struct {
 	pb.PserverServer
+	Param *Parameter
+	Opt   Optimizer
+	PSID  int32
+	lock  *sync.Mutex
 }
 
-func (s *psServer) PullVariable(ctx context.Context, in *pb.PullVariableRequest) (*pb.PullVariableResponse, error) {
-	// TODO: implement the service.
-	return &pb.PullVariableResponse{}, nil
+// NewServer creates a Server instance
+func NewServer(PSID int32, opt string, lr float32) *Server {
+	var ps Server
+	ps.Param = NewParameter()
+	if opt == "SGD" {
```

Comment:
Better to use a function to create an optimizer according to `opt` and other arguments.

Reply:
I have set up an [issue](https://github.com/sql-machine-learning/elasticdl/issues/1703). @LiMinghao1994 is working on this.